Controller/RoleController.php

```diff
@@ -73,7 +73,7 @@ public function viewRoleAction($roleId)
         } catch (NotFoundException $e) {
             $this->notifyError('role.error.role_not_found', [], 'role');
 
-            return $this->redirect($this->generateUrl('admin_roleList'));
+            return $this->pjaxRedirect($this->generateUrl('admin_roleList'));
```

Comment:
OK, but then, as discussed yesterday on Skype, 404 messages should be reported by the JS app as well (see the error notified on line 74).